tests/unit/test_charm.py

```diff
@@ -0,0 +1,128 @@
+# Copyright 2024 mark
+# See LICENSE file for licensing details.
+#
+# Learn more about testing at: https://juju.is/docs/sdk/testing
+
+import unittest
+from unittest.mock import call, patch
+
+import ops
+import ops.testing
+from charm import SdcoreGnbIntegratorCharm
+from ops.model import ActiveStatus, BlockedStatus
+
+GNB_IDENTITY_LIB_PATH = "charms.sdcore_gnbsim_k8s.v0.fiveg_gnb_identity"
+
+
+class TestCharm(unittest.TestCase):
+    def setUp(self):
+        self.namespace = "whatever"
+        self.harness = ops.testing.Harness(SdcoreGnbIntegratorCharm)
+        self.harness.set_model_name(name=self.namespace)
+        self.addCleanup(self.harness.cleanup)
+        self.harness.begin()
+
+    def test_given_default_config_when_config_changed_then_status_is_active(self):
+        self.harness.evaluate_status()
+        self.harness.update_config(key_values={})
+        self.assertEqual(self.harness.charm.unit.status, ActiveStatus())
+        self.harness.charm.on.update_status.emit()
+
+    @patch(f"{GNB_IDENTITY_LIB_PATH}.GnbIdentityProvides.publish_gnb_identity_information")
+    def test_given_fiveg_gnb_identity_relation_created_then_gnb_name_and_tac_are_published(
+        self, patched_publish_gnb_identity
+    ):
+        self.harness.set_leader(is_leader=True)
+
+        test_tac = "012"
+        test_tac_int = 18
+        expected_gnb_name = f"{self.namespace}-gnb-{self.harness.charm.app.name}"
+        self.harness.update_config(key_values={"tac": test_tac})
+
+        relation_id = self.harness.add_relation("fiveg_gnb_identity", "gnb_identity_requirer_app")
+        self.harness.add_relation_unit(relation_id, "gnb_identity_requirer_app/0")
+
+        patched_publish_gnb_identity.assert_called_once_with(
+            relation_id=relation_id, gnb_name=expected_gnb_name, tac=test_tac_int
+        )
+
+    @patch(f"{GNB_IDENTITY_LIB_PATH}.GnbIdentityProvides.publish_gnb_identity_information")
+    def test_given_no_tac_in_config_when_fiveg_gnb_identity_relation_is_added_then_default_tac_is_published(
```

Comment:
Do we need this test? It seems to be checking the same thing as the one above. It's also not exactly true that there's no tac in the config - as long as we have the default value, there's always gonna be something.

Reply:
I think `test_given_fiveg_gnb_identity_relation_created_then_gnb_name_and_tac_are_published` also covers this test as default tag will be published as it is not configured.

Reply:
It is to show that the charm does work even when there is no default set. The test Gulsum mentioned only covers if the user changes the config to show that a new relation event is published with the correct info.